components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/common/SynapsePropertiesHandler.java

```diff
@@ -30,6 +34,23 @@ public boolean handleRequest(MessageContext messageContext) {
         messageContext.setProperty("keyManager.port",mgtHttpsPort);
         String keyManagerHost = System.getProperty(APIConstants.KEYMANAGER_HOSTNAME);
         messageContext.setProperty("keyManager.hostname",keyManagerHost);
+
+        String httpMethod = (String) ((Axis2MessageContext) messageContext).getAxis2MessageContext().
+                getProperty(Constants.Configuration.HTTP_METHOD);
+        org.apache.axis2.context.MessageContext axis2MsgContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        Map headers =
+                (Map) (axis2MsgContext).getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+        boolean isContentTypeNotSet = false;
+        if (headers != null) {
+            isContentTypeNotSet = headers.get("Content-Type") == null || headers.get("Content-Type").equals("");
```

Comment:
Can we use StringUtils.isBlank() if possible?

Reply:
StringUtils.isBlank(..) or isEmpty(..) methods expect String arguments and here headers Map's get(..)  returns an Object. So if we want to use before mentioned methods, we will have to cast the get(..) return values to String and that is not a meaningful usage.

Reply:
noted.